### ISSUE TITLE
Activity of this project

### ISSUE THREAD

**Issue by FancyChaos** [flyte/upnpclient#29]
Hello guys and especially @flyte

Lately I experimented with UPnP stuff and found this library which already does a lot of stuff I needed.
Sadly the project seems a little bit stalled, the last official release was at the beginning of 2018 (!) and code improvements are very rare.
I found quite a few bugs and weird code choices in the current **develop** branch of this project and have to ask the question if this project is currently, actively maintained?

I would love to tackle some issues and bugs in this project, but if this repository is basically dead, I would just develop stuff on my own fork of this project and create my own PIP entry.

Long story short: What is the future of this project?

**Comment by flyte** [comment]
Hi @FancyChaos 

Yep, to be honest I'm not using the library on any projects at the moment, so I'm not very inclined to dedicate much time to it.

I'd be happy to add you as a contributor and set up a way of releasing new versions if you're interested in doing so. You're also absolutely welcome to fork, but at the moment the PyPI project will still be built from this repo. That being said, if you were to make really meaningful updates (perhaps including docs etc.) then I wouldn't want to stand in the way of real progress.

If you want to start with pull requests, then I'll try to be responsive enough to discuss and merge them in.

Cheers

**Comment by FancyChaos** [comment]
As long you are available to approve/merge pull requests and release new versions from time to time I'm more than happy :)

I'll start with fixing the major bugs I came across as well as cleaning up some weird code. Maybe fixing the examples later on and stuff like that.

This issue can be seen as closed or you keep it open so others can see what the current status of this project is, your call.

Have a nice day! (or night^^)

**Comment by flyte** [comment]
Thanks for the latest contributions. Your good explanations make the changes much easier for me to review and merge.

Please make a new Issue if you want me to cut a new release, and I'll try to get around to it as quickly as I can.

**Comment by JarbasAl** [comment]
i see that latest release on pip is 0.0.8 and github is already at 1.0.2 any ETA on a new pip release?

I'm using this in a project and would like to bump versions

**Comment by flyte** [comment]
> any ETA on a new pip release?

Oops, I released a new version on PyPI the same day that you asked for it, but it looks like I forgot to reply.

In case you hadn't already figured it out, @JarbasAl the PyPI version is now 1.0.3.

**Comment by flyte** [comment]
The project has been brought up to date as of v2.0.0, well overdue. Modernised tooling (uv, hatchling, GitHub Actions), updated dependencies, and now supports Python 3.9-3.13.
